### PR TITLE
Local Reporting

### PR DESCRIPTION
**Pull request by alexfornuto**
## Summary

**[Reports](localhost:8000/reports)** (Local development instances only) - Creates a reporting page, with reports on docs with review dates.

## Remaining Work

The following changes still need to be completed:

- [ ] Additional reports?
- [ ] Technical Review

## Post Launch

- [ ] Update Status Report
- [ ] Remove from the [project board](https://github.com/pantheon-systems/documentation/projects/14)

### CODE REVIEW

**Comment by pantheondocs** [comment]
Lighthouse Audit Results:

| Description | Live | pr-5618 | Difference |
| --- | --- | --- | --- |
| Performance Score | `59` | `42` | `-17` |
| Best Practice Score | `86` | `86` | `0` |
| SEO Score | `96` | `96` | `0` |
| Accessibility Score | `96` | `96` | `0` |

Lighthouse Audit Reports:

- [`pr-5618` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/19587/artifacts/0//lighthouse_data/lighthouse-audit-dev.html)
- [`Pantheon Docs` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/19587/artifacts/0//lighthouse_data/lighthouse-audit-reference.html)

**Comment by pantheondocs** [comment]
Lighthouse Audit Results:

| Description | Live | pr-5618 | Difference |
| --- | --- | --- | --- |
| Performance Score | `55.00000000000001` | `52` | `-3` |
| Best Practice Score | `86` | `86` | `0` |
| SEO Score | `96` | `96` | `0` |
| Accessibility Score | `96` | `96` | `0` |

Lighthouse Audit Reports:

- [`pr-5618` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/19592/artifacts/0//lighthouse_data/lighthouse-audit-dev.html)
- [`Pantheon Docs` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/19592/artifacts/0//lighthouse_data/lighthouse-audit-reference.html)

File: gatsby/src/reports/reports.js [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+                  frontmatter {
+                    title
+                    reviewed
+                  }
+                  fields {
+                    slug
+                  }
+                }
+              }
+            }
```
Comment (alexfornuto) [inline gatsby/src/reports/reports.js:49]: This query, titled `staleDocs`, returns the same values as above, but only for pages with a `reviewed` date earlier than `2020-01-01`.

File: gatsby/src/reports/reports.js [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+          }
+        `}
```
Comment (alexfornuto) [inline gatsby/src/reports/reports.js:70]: The query `unreviewedDocs` returns the same values as above, for pages that  have a title (excluding partials) but don't have a reviewed value.

File: gatsby/src/reports/reports.js [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+        render={data => {
+          const [search, setSearch] = useState("")
```
Comment (alexfornuto) [inline gatsby/src/reports/reports.js:72]: This constant uses the useState react hook to capture input from the user.

File: gatsby/src/reports/reports.js [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+          )
+          //console.log("Tertiary Pages: ", tertiaryPages) // For debugging
+          const oldPages = data.staleDocs.edges
+          const unreviewedPages = data.unreviewedDocs.edges
+          //console.log("Unreviewed Docs Array: ", unreviewedPages) // For debugging
+          const unreviewedTertiaryPages = data.unreviewedDocs.edges.filter(
+            page => {
+              return page.node.fields.slug.match(/\/guides(\/[a-z,\-]*){2}/)
+            }
+          )
```
Comment (alexfornuto) [inline gatsby/src/reports/reports.js:87]: These constants refer to sub-arrays from the queries above, reducing the amount of drilling down we have to repeat. Additionally, the `Tertiary` constants filter for just those pages with slugs that match the "/guides/*/*` pattern. This is used to differentiate between all docs and guides, and the pages of a multipage guide which we may count as a single entry for some reporting.

File: gatsby/src/reports/reports.js [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+                  <div
+                    style={{ background: "#fff; cursor:pointer" }}
+                    className="input-group-addon"
+                    id="clear-filter"
+                    onClick={e => setSearch("")}
+                  >
+                    <span className="fa fa-times" />
+                  </div>
+                </div>
+              </div>
```
Comment (alexfornuto) [inline gatsby/src/reports/reports.js:115]: This section was shamelessly stolen from `/terminus/commands`, and creates a form for users to enter a page name and filter the results.

**Pull request merged**